merkle/sparse_merkle_tree.go

```diff
@@ -230,10 +234,15 @@ func (s *subtreeWriter) buildSubtree(ctx context.Context, queueSize int) {
 					NodeRevision: s.treeRevision,
 				})
 		}
+		// Prewarm the cache:
+		var err error
+		_, err = tx.GetMerkleNodes(ctx, s.treeRevision, sibs)
+		if err != nil {
+			return fmt.Errorf("failed to preload node hash cache: %s", err)
+		}
 
 		// calculate new root, and intermediate nodes:
 		hs2 := NewHStar2(s.treeID, s.hasher)
-		var err error
 		root, err = hs2.HStar2Nodes(s.prefix, s.subtreeDepth, leaves,
 			func(depth int, index *big.Int) ([]byte, error) {
```

Comment:
Is this function needed now, as all the nodes are pre-cached? Maybe always return an error from it, to verify that it's indeed **all** the nodes.

Reply:
I'm wrong, the function still needs to return correct data.